modules/host/network.go

```diff
@@ -49,11 +49,11 @@ func (h *Host) initNetworking(address string) (err error) {
 		return err
 	}
 	// Automatically close the listener when h.tg.Stop() is called.
+	err = h.tg.Add()
+	if err != nil {
+		return err
+	}
 	go func() {
-		err := h.tg.Add()
-		if err != nil {
-			return
-		}
```

Comment:
What I'm seeing in the dump:

+ thread frozen on host.go:310 (`tg.Stop()`)
+ thread frozen on network.go:178 - looks like the file changed b/c network.go:178 in this PR points to the start of the function. Is it blocking on `tg.Add()` or another call?

If it's closing on `tg.Add()` it's an implementation error in `tg`, because `tg` should never hang on `tg.Add()`.